toonz/sources/common/timage_io/timage_io.cpp

```diff
@@ -631,6 +631,9 @@ void TImageWriter::save(const TImageP &img) {
     writer->open(file, info);
 
     ras->lock();
+    if (m_path.getType() == "png") {
+      TRop::depremultiply(ras);
+    }
```

Comment:
How about bringing this depremultiply process into the `PngWriter::writeLine()` function?

For now, there seems to be two problems as follows:
- If you activate `Open Flipbook after Rendering` preference option, rendered result shown in the flipbook looks strange as it is depremultiplied.
- For now the image is depremultiplied even if the `Alpha Channel` option of the PNG file format is off. In such case the rendered result becomes incorrect.

I think moving the depremultiply process into `PngWriter::writeLine()` can solve these problems and IMO it would be more reasonable to handle a format-specific process like this in each format's `Tiio::Writer` .

Reply:
I tried playing with this and I can't get the correct results that way.  I just don't know enough about what I'm doing to get it right.

Reply:
I've tried playing with this and I can't seem to get the right results this way.  I don't know enough about premultiplying to get it working.

Reply:
I'll try to modify this.